parser/src/test/java/com/facebook/coresql/parser/sqllogictest/java/SqlLogicTest.java

```
@@ -0,0 +1,336 @@
+            throws IOException
+    {
+        List<String> lines = readAllLines(file);
+        List<List<String>> statementsAsListsOfLines = getListOfStatementsAsLines(lines);
+        return statementsAsListsOfLines.stream()
+                .map(this::buildSqlLogicTestStatementFromLines)
+                .filter(this::statementIsAcceptableToCurrDialect)
+                .collect(toList());
+    }
+
+    private List<SqlLogicTestStatement> getErroneousStatements(List<SqlLogicTestStatement> testStatements)
+    {
+        return testStatements.stream()
+                .filter(statement -> checkStatementForParserError(statement.getStatement()).isPresent())
+                .collect(toList());
+    }
+
+    private class SqlLogicTestFileVisitor
+            implements FileVisitor<Path>
+    {
+        @Override
+        public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs)
+        {
+            return CONTINUE;
+        }
+
+        @Override
+        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
+                throws IOException
+        {
+            if (isHidden(file)) {
+                return CONTINUE;
+            }
+            if (result.getNumFilesVisited() >= maxNumFilesToVisit) {
+                return TERMINATE;
+            }
+            result.incrementNumFilesVisited();
+            List<SqlLogicTestStatement> testStatements = parseSqlLogicTestStatementsFromFile(file);
+            result.incrementNumStatementsFound(testStatements.size());
+            List<SqlLogicTestStatement> erroneousStatements = getErroneousStatements(testStatements);
+            erroneousStatements.forEach(result::addErroneousStatement);
+            return CONTINUE;
+        }
+
+        @Override
+        public FileVisitResult visitFileFailed(Path file, IOException exc)
+        {
+            LOG.warn("Failed to access file: " + file.toString());
```

Don't swallow the exception here, log it so its easier to debug 
nit: use a format string rather than concat

---

Done.